cli/cmd/recover.go

```diff
@@ -0,0 +1,210 @@
+	kdsGetter := snp.NewCachedHTTPSGetter(kdsCache, snp.NeverGCTicker, log.WithGroup("kds-getter"))
+	validator := snp.NewValidator(validateOptsGen, kdsGetter, log.WithGroup("snp-validator"))
+	dialer := dialer.NewWithKey(atls.NoIssuer, validator, &net.Dialer{}, workloadOwnerKey)
+
+	log.Debug("Dialing coordinator", "endpoint", flags.coordinator)
+	conn, err := dialer.Dial(cmd.Context(), flags.coordinator)
+	if err != nil {
+		return fmt.Errorf("dialing coordinator: %w", err)
+	}
+	defer conn.Close()
+
+	client := recoveryapi.NewRecoveryAPIClient(conn)
+	req := &recoveryapi.RecoverRequest{
+		Seed: seed,
+		Salt: salt,
+	}
+	if _, err := client.Recover(cmd.Context(), req); err != nil {
+		return fmt.Errorf("recovering: %w", err)
+	}
+	log.Debug("Got response")
+
+	fmt.Fprintln(cmd.OutOrStdout(), "✔️ Successfully recovered the Coordinator")
+	return nil
+}
+
+type recoverFlags struct {
+	coordinator           string
+	policy                []byte
+	seedSharesFilename    string
+	seedShareOwnerKeyPath string
+	workloadOwnerKeyPath  string
+	manifestPath          string
+}
+
+func decryptedSeedFromShares(seedSharesPath, seedShareOwnerKeyPath string) ([]byte, []byte, error) {
+	key, err := loadSeedshareOwnerKey(seedShareOwnerKeyPath)
+	if err != nil {
+		return nil, nil, err
+	}
+	pub, ok := key.Public().(*rsa.PublicKey)
+	if !ok {
+		return nil, nil, fmt.Errorf("could not get public key from seedshare owner key")
+	}
+	pubHexStr := manifest.MarshalSeedShareOwnerKey(pub).String()
+	var seedShareDoc userapi.SeedShareDocument
+	seedShareBytes, err := os.ReadFile(seedSharesPath)
+	if err != nil {
+		return nil, nil, fmt.Errorf("reading seed shares: %w", err)
+	}
+	if err := json.Unmarshal(seedShareBytes, &seedShareDoc); err != nil {
+		return nil, nil, fmt.Errorf("unmarshaling seed shares: %w", err)
+	}
+	for _, share := range seedShareDoc.SeedShares {
+		if share.PublicKey != pubHexStr {
+			continue
+		}
+		seed, err := manifest.DecryptSeedShare(key, share)
+		if err != nil {
+			return nil, nil, fmt.Errorf("decrypting seed share: %w", err)
+		}
+		return seed, seedShareDoc.Salt, nil
+	}
+	return nil, nil, fmt.Errorf("no matching seed share found")
+}
+
+func parseRecoverFlags(cmd *cobra.Command) (recoverFlags, error) {
+	coordinator, err := cmd.Flags().GetString("coordinator")
+	if err != nil {
+		return recoverFlags{}, err
+	}
+	policy, err := decodeCoordinatorPolicyHash(cmd.Flags())
+	if err != nil {
+		return recoverFlags{}, err
+	}
+	seed, err := cmd.Flags().GetString("seed")
+	if err != nil {
+		return recoverFlags{}, err
+	}
+	seedShareOwnerKeyPath, err := cmd.Flags().GetString("seedshare-owner-key")
+	if err != nil {
+		return recoverFlags{}, err
+	}
+	workloadOwnerKeyPath, err := cmd.Flags().GetString("workload-owner-key")
+	if err != nil {
+		return recoverFlags{}, err
+	}
+	manifestPath, err := cmd.Flags().GetString("manifest")
+	if err != nil {
+		return recoverFlags{}, err
+	}
+
+	return recoverFlags{
```

Comment:
Can you prepend the workspace directory here if it is set, like in the other commands?